src/server/main_service.cc

```diff
@@ -1441,7 +1456,13 @@ void Service::EvalInternal(const EvalArgs& eval_args, Interpreter* interpreter,
   interpreter->SetGlobalArray("ARGV", eval_args.args);
   interpreter->SetRedisFunc([cntx, this](auto args) { CallFromScript(cntx, args); });
 
-  Interpreter::RunResult result = interpreter->RunFunction(eval_args.sha, &error);
+  Interpreter::RunResult result;
+  optional<ShardId> sid = GetSingleShardIdForKeys(*sinfo);
+  if (sid.has_value()) {
+    pp_.at(*sid)->Await([&]() { result = interpreter->RunFunction(eval_args.sha, &error); });
```

Comment:
@chakaz can you please loadtest the server on 16-threads (4xlarge) and attach here the profilez graph?
also, can you say how much CPU utilization approximately it reaches at peak during the load?
I am afraid we spend lots of CPU on launching a new fiber for every `Await` call, and I would like to check it.

Reply:
Sure thing, but won't happen today I'm afraid :sleepy: